tests/testthat/test-extract_features.R

```diff
@@ -1,33 +1,96 @@
+    }
 
-  actual <- dplyr::arrange_at(
-    actual, keys
-  )
-  expected <- dplyr::arrange_at(
-    expected, keys
-  )
+    if (!is(cluster, "cluster")) {
+      cluster <- parallel::makeCluster(cluster)
+      on.exit(parallel::stopCluster(cluster))
+    }
+
+    actual <- extract_features(
+      filenames,
+      cluster = cluster,
+      min_pres = min_pres,
+      min_run = min_run,
+      mz_tol = tol,
+      intensity_weighted = intensity_weighted,
+      sd_cut = sd_cut,
+      sigma_ratio_lim = sigma_ratio_lim,
+      baseline_correct = 0,
+      baseline_correct_noise_percentile = 0.05,
+      min_bandwidth = NA,
+      max_bandwidth = NA,
+      moment_power = 1,
+      BIC_factor = 2.0,
+      component_eliminate = 0.01,
+      peak_estim_method = "moment",
+      shape_model = "bi-Gaussian"
+    )
+
+    expected_filenames <- lapply(files, function(x) {
+      file.path(testdata, x, paste0(x, "_features.Rds"))
+    })
+
+    expected <- lapply(expected_filenames, readRDS)
+
+    actual <- unique(actual)
+    expected <- unique(expected)
 
-  report <- dataCompareR::rCompare(actual, expected, keys = keys, roundDigits = 4, mismatches = 100000)
-  dataCompareR::saveReport(report, reportName = "extract_features_0", showInViewer = FALSE, HTMLReport = FALSE, mismatchCount = 10000)
+    keys <- c("mz", "pos", "sd1", "sd2")
 
-  expect_true(nrow(report$rowMatching$inboth) >= 0.9 * nrow(expected))
+    actual <- lapply(actual, function(x) {
+      as.data.frame(x) |> dplyr::arrange_at(keys)
+    })
+    expected <- lapply(expected, function(x) {
+      as.data.frame(x) |> dplyr::arrange_at(keys)
+    })
 
-  incommon <- as.numeric(rownames(report$rowMatching$inboth))
 
-  subset_actual <- actual %>% dplyr::slice(incommon)
-  subset_expected <- expected %>% dplyr::slice(incommon)
+    for (i in seq_along(files)) {
+      actual_i <- actual[[i]]
+      expected_i <- expected[[i]]
 
-  expect_equal(subset_actual$area, subset_expected$area, tolerance = 0.01 * max(expected$area))
-})
+      report <- dataCompareR::rCompare(actual_i, expected_i, keys = keys, roundDigits = 4, mismatches = 100000)
+      dataCompareR::saveReport(report, reportName = files[[i]], showInViewer = FALSE, HTMLReport = FALSE, mismatchCount = 10000)
+
+      expect_true(nrow(report$rowMatching$inboth) >= 0.9 * nrow(expected_i))
+
+      incommon <- as.numeric(rownames(report$rowMatching$inboth))
+
+      subset_actual <- actual_i %>% dplyr::slice(incommon)
+      subset_expected <- expected_i %>% dplyr::slice(incommon)
+
+      expect_equal(subset_actual$area, subset_expected$area, tolerance = 0.01 * max(subset_expected$area))
+    }
+  },
+  patrick::cases(
+    RCX_shortened = list(
+      files = c("RCX_01_shortened_v2", "RCX_09_shortened_v2", "RCX_16_shortened_v2"),
+      tol = 1e-06,
+      min_pres = 0.7,
+      min_run = 4,
+      intensity_weighted = TRUE,
+      sd_cut = c(0.01, 500),
+      sigma_ratio_lim = c(0.01, 100)
+    )
```

Comment:
Are we planning to add more test cases?

Reply:
Eventually - I mean it doesn't hurt to already have it in a parameterized form!